pkg/controller/context.go

```diff
@@ -330,17 +339,17 @@ func (c *ContextFactory) Build(component ...string) (*Context, error) {
 	// Create event broadcaster.
 	// Add cert-manager types to the default Kubernetes Scheme so Events can be
 	// logged properly.
-	cmscheme.AddToScheme(scheme.Scheme)
-	gwscheme.AddToScheme(scheme.Scheme)
```

Comment:
Is this the key change? 
I can see in your -race output that this is where the global Kubernetes scheme is being mutated,
via an `init()` function, while another routine is reading the Kubernetes scheme.

Reply:
Good catch, if we just replace the write, the bug is fixed.
I'll just do this change and run the race test again.